orttraining/orttraining/python/training/orttrainer_options.py

```diff
@@ -277,7 +277,19 @@ class ORTTrainerOptions(object):
                             'default' : True
                         }
                     }
-                }
+                },
+                'provider_options':{
+                    'type': 'dict',
+                    'default_setter': lambda _: {},
+                    'required': False,
+                    'allow_unknown': True,
+                    'schema': {}
+                },
+                'session_options': {
+                    'type': 'session_options',
+                    'nullable': True,
+                    'default': None
+                },
```

Comment:
```suggestion
                'provider_options':{
                    'type': 'dict',
                    'default': {},
                    'required': False,
                    'schema': {}
                },
                'session_options': {
                    'type': 'SessionOptions',
                    'nullable': True,
                    'default': None
                },
```
This is a user-friendly schema description. We hide internals